tests/queries/0_stateless/03034_ddls_and_merges_with_unusual_maps.sql

```diff
@@ -0,0 +1,32 @@
+-- Tests maps with "unusual" key types (Float32, Nothing, LowCardinality(String))
+
+SET mutations_sync = 2;
+
+DROP TABLE IF EXISTS tab;
+
+SELECT 'Map(Nothing, ...) is non-comparable --> not usable as primary key';
+CREATE TABLE tab (m1 Map(Nothing, String)) ENGINE = MergeTree ORDER BY m1; -- { serverError DATA_TYPE_CANNOT_BE_USED_IN_KEY }
+
+SELECT 'But Map(Nothing, ...) can be a non-primary-key, it is quite useless though ...';
+CREATE TABLE tab (m3 Map(Nothing, String)) ENGINE = MergeTree ORDER BY tuple();
+-- INSERT INTO tab VALUES (map('', 'd')); -- { serverError NOT_IMPLEMENTED } -- <-- for some weird reason the test won't let me set an expected error
```

Comment:
It is strange. It looks good, so I will have a quick look.

Reply:
Thanks for checking. I now also see that the query doesn't execute the query ... well, the server logs `QUERY_WAS_CANCELLED` but that can also not be checked via `serverError` 😢 

Anyways, we are testing a weird edge case anyways, and the effort to test through a HTTP client or something else is really not worth the trouble. I clarified the comment a bit. It's not pretty but at least it is documented now :-)